[RFC] Resolve FIXMEs and TODOs

Hjdskes opened the pull request:
Left to look at:
- `org.metaborg.spoofax.shell.eclipse`:

```
src/main/java/org/metaborg/spoofax/shell/client/eclipse/impl/EclipseDisplay.java:    // TODO: Use ReplDocument to provide custom partitioning? Perhaps more something for the output
src/main/java/org/metaborg/spoofax/shell/client/eclipse/impl/EclipseDisplay.java:                // TODO: this could mess with the styleranges, perhaps style() shouldn't be called
src/main/java/org/metaborg/spoofax/shell/client/eclipse/impl/EclipseEditor.java:        // TODO: text has been modified, send it to get syntax highlighting.
src/main/java/org/metaborg/spoofax/shell/client/eclipse/impl/EclipseRepl.java:        // TODO: Style input! Output cannot be styled since there is no way to "pretty-prettyprint"
```

Hjdskes committed:
ICommandInvoker: s/FIXME/TODO/

Hjdskes committed:
Console: make the argument work again

Hjdskes committed:
ConsoleReplModule: remove unused provider method

Hjdskes commented on org.metaborg.spoofax.shell.console/src/main/java/org/metaborg/spoofax/shell/client/console/Main.java:
```diff
@@ -15,18 +15,29 @@
+    private static StyledText error(String[] args) {
+        StringBuilder invalidArgs = new StringBuilder();
+        for (String arg : args) {
+            invalidArgs.append(arg);
```
The formatting here has to change still.

Hjdskes committed:
Core: add more supported filesystems to load languages from

Hjdskes commented on org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/commands/LanguageCommand.java:
```diff
@@ -27,6 +27,8 @@
  */
 public class LanguageCommand implements IReplCommand {
+    private static final String[] ARCHIVES =
+        { ".zip", ".jar", ".tar", ".tgz", ".tbz2", };
```
Ideally there should be a test for all of these

Hjdskes replied:
See aeaa3ace5bd96e14dd86b11947a17b464f92fa9e for an attempt. @gfokkema do you think this is enough? Should there be a test for simply having folders too, or any other supported file system?

Unia commented:
I'll leave the hardcoded project directory FIXMEs, as there is no solution to these yet. Even so, I think the current situation is acceptable.

Hjdskes committed:
EclipseDisplay: remove outdated TODO

Hjdskes committed:
ColorManager: remove unused getDefault method

gfokkema committed:
Fix most / all javadoc errors.

Hjdskes committed:
LanguageCommandTest: test all supported archives parameterized

Hjdskes committed:
Eclipse: style changes

EclipseEditor: monospace font and scrollbars when needed
EclipseDisplay: user input is now bold as opposed to having a weird blue
                background.

Hjdskes committed:
Main: fix the error formatting

Hjdskes committed:
Core: fix more javadoc

Hjdskes committed:
Console: fix javadoc errors

gfokkema committed:
Add a defaultcommand to the invoker.

Hjdskes commented on org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/ReplModule.java:
```diff
@@ -62,7 +62,8 @@
         commandBinder.addBinding("load").to(LanguageCommand.class);
-
+        bind(IReplCommand.class).annotatedWith(Names.named("default_command"))
+        .to(HelpCommand.class);
```
Instead of binding HelpCommand you could create a dummy implementation in a lambda with a more accurate error message?

gfokkema replied:
No that's not possible.

gfokkema replied:
Well, an anonymous instance _is_ possible, but very ugly xD

Hjdskes commented on org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/invoker/ICommandInvoker.java:
```diff
@@ -49,7 +60,7 @@ default IResult execute(String optionallyPrefixedCommandName)
             return commandFromName(commandName).execute(argument);
         } else {
             // TODO: create sensible way to set default
```
This can be removed now :P

gfokkema committed:
Create a DefaultCommand class and bind to that class.

Hjdskes committed:
Core: Fixes from the IntelliJ static analyzer

Hjdskes committed:
Eclipse: Fixes from the IntelliJ static analyzer

Hjdskes committed:
Console: Fixes from the IntelliJ static analyzer

Hjdskes committed:
Fix LanguageCommand not loading archives

Hjdskes committed:
ICommandInvoker & EclipseRepl: remove leftover TODO
